components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/mediators/AWSLambdaClassMediator.java

```diff
@@ -0,0 +1,171 @@
+
+        String payload;
+        if (JsonUtil.hasAJsonPayload(axis2MessageContext)) {
+            payload = JsonUtil.jsonPayloadToString(axis2MessageContext);
+        } else {
+            payload = "{}";
+        }
+
+        InvokeResult invokeResult = invokeLambda(payload);
+
+        if (invokeResult != null) {
+            if (log.isDebugEnabled()) {
+                log.debug("AWS Lambda function: " + resourceName + " is invoked successfully.");
+            }
+            JsonUtil.setJsonStream(axis2MessageContext, new ByteArrayInputStream(invokeResult.getPayload().array()));
+            axis2MessageContext.setProperty(APIMgtGatewayConstants.HTTP_SC, invokeResult.getStatusCode());
+            axis2MessageContext.setProperty(APIMgtGatewayConstants.REST_MESSAGE_TYPE, APIConstants.APPLICATION_JSON_MEDIA_TYPE);
+            axis2MessageContext.setProperty(APIMgtGatewayConstants.REST_CONTENT_TYPE, APIConstants.APPLICATION_JSON_MEDIA_TYPE);
+            axis2MessageContext.removeProperty(APIConstants.NO_ENTITY_BODY);
+        } else {
+            if (log.isDebugEnabled()) {
+                log.debug("Failed to invoke AWS Lambda function: " + resourceName);
+            }
+            axis2MessageContext.setProperty(APIMgtGatewayConstants.HTTP_SC, APIMgtGatewayConstants.HTTP_SC_CODE);
+            axis2MessageContext.setProperty(APIConstants.NO_ENTITY_BODY, true);
+        }
+
+        return true;
+    }
+
+    /**
+     * invoke AWS Lambda function
+     * @param payload - input parameters to pass to AWS Lambda function as a JSONString
+     * @return InvokeResult
+     */
+    private InvokeResult invokeLambda(String payload) {
+        try {
+            AWSCredentialsProvider credentialsProvider;
+            if (StringUtils.isEmpty(accessKey) && StringUtils.isEmpty(secretKey)) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Using temporary credentials supplied by the IAM role attached to the EC2 instance");
+                }
+                credentialsProvider = InstanceProfileCredentialsProvider.getInstance();
+            } else if (!StringUtils.isEmpty(accessKey) && !StringUtils.isEmpty(secretKey)) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Using user given stored credentials");
+                }
+                if (secretKey.length() == APIConstants.AWS_ENCRYPTED_SECRET_KEY_LENGTH) {
+                    CryptoUtil cryptoUtil = CryptoUtil.getDefaultCryptoUtil();
+                    setSecretKey(new String(cryptoUtil.base64DecodeAndDecrypt(secretKey),
+                            APIConstants.DigestAuthConstants.CHARSET));
+                }
+                BasicAWSCredentials awsCredentials = new BasicAWSCredentials(accessKey, secretKey);
+                credentialsProvider = new AWSStaticCredentialsProvider(awsCredentials);
+            } else {
+                throw new SdkClientException("Missing AWS Credentials");
```

Comment:
this exception is caught in this method itself. In that case, do we really need to throw an exception?

Reply:
logged error instead throwing an error